applications/DEMApplication/python_scripts/sphere_strategy.py

```diff
@@ -319,12 +320,41 @@ def SolveSolutionStep(self):
         self.FixExternalForcesManually(time)
         (self.cplusplus_strategy).Solve()
 
-    def AdvanceInTime(self, step, time):
+    def AdvanceInTime(self, step, time, is_time_to_print = False):
         step += 1
-        time = time + self.dt
-
+        time += self.dt
+        self.UpdateTimeInModelParts(time, step, is_time_to_print)
```

Comment:
UpdateTimeInModelParts is a protected method. It should start with _ to reflect that